guide/src/reference/arbitrary-data-with-serde.md

```diff
@@ -94,23 +92,61 @@ example.field2.push([5, 6]);
 receive_example_from_js(example);
 ```
 
-## An Alternative Approach: `serde-wasm-bindgen`
-
-[The `serde-wasm-bindgen`
-crate](https://github.com/cloudflare/serde-wasm-bindgen) serializes and
-deserializes Rust structures directly to `JsValue`s, without going through
-temporary JSON stringification. This approach has both advantages and
-disadvantages.
-
-The primary advantage is smaller code size: going through JSON entrenches code
-to stringify and parse floating point numbers, which is not a small amount of
-code. It also supports more types than JSON does, such as `Map`, `Set`, and
-array buffers.
-
-There are two primary disadvantages. The first is that it is not always
-compatible with the default JSON-based serialization. The second is that it
-performs more calls back and forth between JS and Wasm, which has not been fully
-optimized in all engines, meaning it can sometimes be a speed
-regression. However, in other cases, it is a speed up over the JSON-based
-stringification, so &mdash; as always &mdash; make sure to profile your own use
-cases as necessary.
+## An alternative approach - using JSON
+
+`serde-wasm-bindgen` works by directly manipulating JavaScript values. This
+requires a lot of calls back and forth between Rust and JavaScript, which can
+sometimes be slow. An alternative way of doing this is to serialize values to
+JSON, and then parse them on the other end. Browsers' JSON implementations are
+usually quite fast, and so this approach can outstrip `serde-wasm-bindgen`'s
+performance in some cases.
```

Comment:
```suggestion
performance in some cases. But this approach supports only types that can be
serialized as JSON, leaving out some important types that `serde-wasm-bindgen`
supports such as `Map`, `Set`, and array buffers.
```
... addressing @RReverser's [point about type support](https://github.com/rustwasm/wasm-bindgen/pull/3031/files#r956780944)

Reply:
Oh, so this edit didn't get in? Can you make a separate PR please?

Reply:
Voilà! https://github.com/rustwasm/wasm-bindgen/pull/3047